central/notifiers/email/email.go

```diff
@@ -550,10 +550,6 @@ func (e *email) getPassword() (string, error) {
 		return e.creds, nil
 	}
 
-	if e.notifier.GetNotifierSecret() == "" {
```

Comment:
Given that we already have a break condition for unauthenticated notifier above at [ln540](https://github.com/stackrox/stackrox/pull/9000/files#diff-5db8a56f8a6d814299518832fa42c46934ea32c5b0bcf9ccca29d31e49b117c8L540) the NotifierSecret should be non-empty if we come here. If it is empty, the decrypt function will eventually error out if it receives empty NotifierSecret.

Reply:
So no change required, we removed the check and it was moot to begin with.